### PR TITLE
Update release.yml

### PR DESCRIPTION
**Pull request by KennethEnevoldsen**
Updated release workflow. 

Tested (and worked) on https://github.com/KennethEnevoldsen/scandinavian-embedding-benchmark

edit: turns out that this does not push to PyPI. Their new approach is def. better (also make the action more flexible) - still experimenting with a solution.

### CODE REVIEW

File: {{cookiecutter.project_name}}/.github/workflows/release.yml [hunk 1]
```diff
@@ -14,24 +15,32 @@ jobs:
...
+        uses: python-semantic-release/python-semantic-release@v8.0.4
+        with:
+          github_token: ${{ secrets.PAT }}
+
+      - name: Publish package distributions to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        if: steps.release.outputs.released == 'true'
+        # This action supports PyPI's trusted publishing implementation, which allows authentication to PyPI without a manually 
+        # configured API token or username/password combination. To perform trusted publishing with this action, your project's 
+        # publisher must already be configured on PyPI.
```
Comment (MartinBernstorff) [inline {{cookiecutter.project_name}}/.github/workflows/release.yml:40]: Oh, my understanding is that users need to do less now, then?

Perhaps we can include a link on how to seutp PyPI to "trust a given GitHub repository and workflow"? https://blog.pypi.org/posts/2023-04-20-introducing-trusted-publishers/

File: {{cookiecutter.project_name}}/pyproject.toml [hunk 2]
```diff
@@ -98,17 +99,9 @@ select = [
...
-ignore = [
-  "ANN101",
-  "ANN401",
-  "E402",
-  "E501",
-  "F401",
-  "F841",
-  "RET504"
+  "W",
 ]
```
Comment (MartinBernstorff) [inline {{cookiecutter.project_name}}/pyproject.toml:103]: Why the formatting changes to the TOML? :-)

**Review by MartinBernstorff — APPROVED** [review]
Looks good! A few tiny questions, then I think it's ready for merge 👍

File: {{cookiecutter.project_name}}/pyproject.toml (hunk 2 above)
Comment (KennethEnevoldsen) [reply {{cookiecutter.project_name}}/pyproject.toml:103]: Probably an error

**Comment by MartinBernstorff** [comment]
I've added it to recommended setup as well. If you agree, feel free to merge 👍

**Pull request merged**